### PR TITLE
get controller tests passing

### PR DESCRIPTION
**Pull request by kelseyhuse30**
This should get all the controller tests passing for now, although there are some pending tests that can be tackled later on.

I used the login_user method at the start of all controllers in order to generate a current_user which makes these requests now valid.  The [devise docs](https://github.com/plataformatec/devise/wiki/How-To:-Test-controllers-with-Rails-(and-RSpec)#rails-3-rspec-scaffolds) recommended removing the `valid_session` boilerplate that rspec generates, since Devise has their own methods.

### CODE REVIEW

**Comment by lortza** [comment]
Awesome! Thanks @kelseyhuse30! I need to spend a little time looking into how in the heck a test can pass after creating an object with `valid_attributes` that are `nil` -- even when that object's class has required fields. Do you know why this works?

**Comment by kelseyhuse30** [comment]
@lortza Hmm weird! I didn't see that happen anywhere, could you point out where you saw it? Most of the valid_attributes tests were skipped by default for now, I only updated one for fun.

**Comment by lortza** [comment]
@kelseyhuse30 Ohhhh I'm seeing it now. I wasn't understanding the scope of the `skip`. But now that I've run the files locally, I get it. Hey thanks for teaching me about controller specs!

**Comment by lortza** [comment]
@kelseyhuse30 Have you ever used factories in your controller specs? I have factories set up (https://github.com/lortza/therapy_tracker/tree/master/spec/factories) that we can use to avoid manually creating all of those valid and invalid attributes. Example:

This:
https://github.com/lortza/therapy_tracker/blob/master/spec/controllers/body_parts_controller_spec.rb#L8-L9
```
let(:valid_attributes) { build(:body_part, user_id: subject.current_user.id).attributes }
let(:invalid_attributes) { build(:body_part, user_id: subject.current_user.id, name: '').attributes }
```

Instead of this:
```
let(:valid_attributes) {
  {
    name: 'elbow',
    user_id: subject.current_user.id
  }
}

let(:invalid_attributes) {
  {
    name: '',
    user_id: subject.current_user.id
  }
}
```

Unless you see otherwise, I'm thinking I'd like to use them in these specs going forward.

**Comment by kelseyhuse30** [comment]
@lortza ah I didn't know about that attributes method! That is awesome.  I'll submit another PR to use that strategy and un-skip the rest of those controller tests. Stay tuned...

File: .travis.yml [hunk 1]
```diff
@@ -13,4 +13,5 @@ script:
   - bundle exec rake db:create
   - bundle exec rake db:migrate
   - bundle exec rspec spec/models
+  - bundle exec rspec spec/controllers
```
Comment (lortza) [inline .travis.yml]: @kelseyhuse30 Thanks for following this pattern! I decided that it was time to run more of the suite, so I ended up doing an exclusion pattern instead of individually including the directories I wanted to run: 
```
bundle exec rspec --exclude-pattern "**/views/*/*_spec.rb, **/requests/*_spec.rb"
```
